lib/syskit/roby_app/plugin.rb

```diff
@@ -146,6 +150,58 @@ def self.setup(app)
 
                 rtt_core_model = app.default_loader.task_model_from_name("RTT::TaskContext")
                 Syskit::TaskContext.define_from_orogen(rtt_core_model, register: true)
+
+                # Log Transfer FTP Server spawned during Application#setup
+                app.setup_local_log_transfer_server if app.log_transfer_ip
+            end
+
+            def setup_local_log_transfer_server
+                @log_transfer_user = "process server"
+                @log_transfer_password = SecureRandom.base64(15)
+
+                tmp_root_ca = TmpRootCA.new(log_transfer_ip)
+                @log_transfer_certificate = tmp_root_ca.certificate
+
+                start_local_log_transfer_server(log_dir, @log_transfer_user, @log_transfer_password, tmp_root_ca.private_certificate_path)
+
+                @log_transfer_port = @log_transfer_server.port
```

Comment:
Sorry ... Didn't explain myself properly.

You should *still* not define `@log_transfer_port` use `@log_transfer_server.port` whenever you need the port...